planning/lifetimes/requirements.md

```
@@ -0,0 +1,1144 @@
+
+```
+val items: mut Array<number> = [1, 2, 3]
+val f = fn() { items.push(4) }  // f captures `items` mutably
+val snapshot: Array<number> = items  // ERROR: f is live and holds a mutable
+                                     // reference to `items`
+f()
+```
+
+If the closure only reads the captured variable, it holds an immutable reference:
+
+```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)